Astro CLI Tab - Overview, Quickstart, and Configure CLI docs (#711)
astronomer/docs: astro/cli/configure-cli.md

---
sidebar_label: 'Configure the CLI'
title: 'Configure the Astro CLI'
id: configure-cli
description: Install, upgrade, and manage settings for the Astro CLI.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

## Install the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To install the latest version of the Astro CLI, run the following command:

```sh
brew install astronomer/cloud/astro
```

To install a specific version of the Astro CLI, specify the version you want to install at the end of the command:

```sh
brew install astronomer/cloud/astro@0.XX
```

</TabItem>

<TabItem value="windows">

1. In a PowerShell terminal, create a new directory for your Astro project and set it as your current directory:

    ```powershell
    mkdir my-project && cd my-project
    ```

> 2. Based on your CPU, run one of the following commands to download the Astro CLI executable into your project directory.

Reviewer: These links only work for pre-releases. Once we release v1 users will need to go to https://github.com/astronomer/astro-cli/releases to find the release they want a download it. For example if they want the release for windows AMD 64 for v0.29.0 they would go to https://github.com/astronomer/astro-cli/releases/tag/v0.29.0 and click on 
astro_0.29.0_windows_amd64.zip  this should automatically unzip the file into there downloads folder next they will add the executable to either their Astro project to run as ./astro or add it to PATH to run globally

    - ARM64:

        <pre><code parentName="pre">{`Invoke-WebRequest -Uri https://goreleaserdev.blob.core.windows.net/goreleaser-test-container/releases/v${siteVariables.cliVersion}/cloud-cli_${siteVariables.cliVersion}_Windows_arm64.tar.gz -OutFile astrocli.tar.gz`}</code></pre>

    To install a different version of the CLI, change the version number in the command to your desired CLI version.

3. Run the following command to unzip the executable:

    ```sh
    tar -xvzf .\astrocli.tar.gz
    ```

4. To run the executable without specifying its file path, save `astro.exe` in a secure location on your machine and add its filepath in the Windows PATH environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).

</TabItem>

<TabItem value="linux">

1. In a Linux terminal, create a new directory for your Astro project and set it as your current directory:

    ```sh
    mkdir my-project && cd my-project
    ```

> 2. Based on your CPU, run one of the following commands to download the Astro CLI executable into your project directory.

Reviewer: Linux and mac users can use can use install.astronomer.io to install the CLI directly to PATH. We should not have users mess with PATH if don't need to. For linux we should just ask users to run

```
curl -sSL install.astronomer.io | sudo bash -s
```

Or ask them to run this command for a specific version

```
curl -sSL https://install.astronomer.io | sudo bash -s -- v1.0.0
```

    - ARM64:

        <pre><code parentName="pre">{`curl https://goreleaserdev.blob.core.windows.net/goreleaser-test-container/releases/v${siteVariables.cliVersion}/cloud-cli_${siteVariables.cliVersion}_Linux_arm64.tar.gz -o astrocli.tar.gz`}</code></pre>

    To install a different version of the CLI, change the version number in the command to your desired CLI version.

3. Run the following command to unzip the executable:

    ```sh
    tar xzf astrocli.tar.gz
    ```

4. To run the executable without specifying its file path, save `astro` in a secure location on your machine and add its filepath in the Linux `$PATH` environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).

</TabItem>

</Tabs>

## Upgrade the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To upgrade the Astro CLI, you can run the same command as you did to install the CLI for the first time:

```sh

> brew install astronomer/cloud/astro

Reviewer: ```suggestion
brew install astronomer/tap/astro
```

Reviewer: This command will give the user the latest version of the CLI

```

</TabItem>

<TabItem value="windows">

> [Upgrade steps for Windows]

Reviewer: To upgrade for windows the user will need to go to https://github.com/astronomer/astro-cli/releases to find the latest release for their os, download it, then replace the old executable in PATH or their project with the new executable

</TabItem>

<TabItem value="linux">

> [Upgrade steps for Linux]

Reviewer: To upgrade to the latest version in linux the user just needs to run

```
curl -sSL install.astronomer.io | sudo bash -s
```

</TabItem>

</Tabs>

## Uninstall the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To uninstall the Astro CLI, run the following command:

```sh

> brew uninstall astronomer/cloud/astro

Reviewer: ```suggestion
brew uninstall astronomer/tap/astro
```

```

</TabItem>

<TabItem value="windows">

> [Upgrade steps for Windows]

Author: @sunkickr Can you help me write upgrade and uninstall steps for both Windows and Linux? Do they just need to delete their existing executable and download a new one?

Reviewer: yes, I add some details in my comment on upgrading windows above

Reviewer: Specifically to uninstall they need to remove the the executable from PATH or the their project directory and delete it

</TabItem>

<TabItem value="linux">

[Upgrade steps for Linux]

</TabItem>

</Tabs>

## Migrate from `astrocloud` to `astro`

Complete this migration if all of the following are true:

- You are an Astro user.
- You used a pre-1.0 version of the Astro CLI. For these versions, the CLI executable was `astrocloud`.

### Step 1: Uninstall the `astrocloud` Executable

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To uninstall the old Astro CLI, run the following command:

```sh
brew uninstall astronomer/cloud/astrocloud
```

</TabItem>

<TabItem value="windows">

> [Uninstall steps for Windows]

Reviewer: Remove astrocloud executable from PATH. Would need to look more into how to do this exactly

</TabItem>

<TabItem value="linux">

> [Uninstall steps for Linux]

Reviewer: Remove the astrocloud executable from PATH

</TabItem>

</Tabs>

## Step 2: Install Astro CLI v1.0+

For more details, see [Install the CLI](configure-cli.md#install-the-cli.md).

## Step 3: Migrate Project Configurations (_Optional_)

For each Astro project where you manually updated the `.astrocloud/config.yaml` file:

1. In your terminal, open your Astro project.

> 2. Run `astro dev init` to generate a new `.astro` directory in your project. This subdirectory might be hidden in graphical file browsers. You can show hidden files using `⌘ + Shift + .` on Mac or by selecting **View > Hidden items** in Windows file explorer.

Reviewer: We should note that when the user runs `astro dev init` they will need to say yes to message asking them if they would like to continue because the directory is not empty

---
sidebar_label: 'Configure the CLI'
title: 'Configure the Astro CLI'
id: configure-cli
description: Install, upgrade, and manage settings for the Astro CLI.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

## Install the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

#### Prerequisites

To use the Astro CLI on Mac, you must have:

- [Homebrew](https://brew.sh/)
- [Docker Desktop](https://docs.docker.com/get-docker/) (v18.09 or higher).

> 
> To install the latest version of the Astro CLI, run the following command:

Reviewer 2: ```suggestion

#### Installation

To install the latest version of the Astro CLI, run the following command:
```

```sh
brew install astronomer/tap/astro
```

To install a specific version of the Astro CLI, specify the version you want to install at the end of the command:

```sh

> brew install astronomer/tap/astro@0.XX

Reviewer 2: ```suggestion
brew install astronomer/tap/astro@<major.minor.patch-version>

If you specify only a major version, the latest minor and patch version that corresponds will be installed automatically.
```

@sunkickr This true?

```

</TabItem>

<TabItem value="windows">

#### Prerequisites

To use the Astro CLI on Windows, you must have:

- [Docker Desktop](https://docs.docker.com/desktop/windows/install/) for windows.
- [Docker Engine](https://docs.docker.com/engine/install/) (v0.13.1 or higher).
- [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) enabled on your local machine.
-  Windows 10 or Windows 11.

#### Installation

> 

Reviewer 2: ```suggestion
```

> 1. Go to the [**Releases** page of the Astro CLI GitHub](https://github.com/astro-projects/astro-cli/releases). Based on your desired CLI version and CPU architecture, download one of the `.zip` files available on this page.

Reviewer 2: ```suggestion
1. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astro-projects/astro-cli/releases). Based on your desired CLI version and CPU architecture, download one of the `.zip` files available on this page.
```

Reviewer 2: What are all the possible values depending on `CPU architecture`? I see `amd64` - is there something else users would pick between? I'd list em out if so

Reviewer 2: Fair enough

>     For example, if you wanted to install v1.0.0 of the Astro CLI on a Windows Machine with an AMD 64 architecture, you would download `astro_1.0.0-converged_windows_amd64.zip`.

Reviewer 2: ```suggestion
    For example, to install v1.0.0 of the Astro CLI on a Windows Machine with an AMD 64 architecture, download `astro_1.0.0-converged_windows_amd64.zip`.
```

2. Run the following command to unzip the executable:

    ```sh
    tar -xvzf .\astrocli.tar.gz
    ```

> 3. Save `astro.exe` in a secure location on your machine and add its filepath in the Windows PATH environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).

Reviewer 2: ```suggestion
3. Save `astro.exe` in a secure location on your machine and add its filepath as a Windows PATH environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).
```

</TabItem>

<TabItem value="linux">

#### Prerequisites

To use the Astro CLI on Linux, you must have:

> - [Docker Engine](https://docs.docker.com/engine/install/) (v0.13.1 or higher).

Reviewer 2: ```suggestion
- [Docker Engine](https://docs.docker.com/engine/install/) (v0.18.9 or higher).
```

@sunkickr This prob needs to be `v18.09` or higher right? I think 0.13.1 was requirement years ago for Software... assume we should bump it up?

#### Installation

Run the following command to install the latest version of the Astro CLI directly to `PATH`:

```sh
curl -sSL install.astronomer.io | sudo bash -s
```

To install a specific version of the CLI, specify the version number as a flag at the end of this command. For example, to install v1.0.0 of the CLI, you would run:

```sh
curl -sSL install.astronomer.io | sudo bash -s -- v1.0.0
```

> 

Reviewer 2: ```suggestion

If you specify only a major version, the latest minor and patch version that corresponds will be installed automatically.
```

Same thing here - @jwitz There might be a better way to phrase this

</TabItem>

</Tabs>

## Upgrade the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

> To upgrade the Astro CLI, you can run the same command as you did to install the CLI for the first time:

Reviewer 2: ```suggestion
To upgrade the Astro CLI to the latest version, you can run the same command as you did to install the CLI for the first time:
```

```sh
brew install astronomer/tap/astro
```

</TabItem>

> <TabItem value="windows">
> 

Reviewer 2: ```suggestion
<TabItem value="windows">

To upgrade the Astro CLI on Windows, follow the steps below.
```

> 1. Remove the existing `astro.exe` file on your machine.

Reviewer 2: ```suggestion
1. Remove the `astro.exe` file on your machine that corresponds to your currently running version.
```

Author: There should only be one of these on their machine at the time, I think adding this extra line will be more confusing than clarifying

> 2. Go to the [**Releases** page of the Astro CLI GitHub](https://github.com/astro-projects/astro-cli/releases). Based on your desired CLI version and CPU architecture, download one of the `.zip` files available on this page.

Reviewer 2: ```suggestion
2. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astro-projects/astro-cli/releases). Based on the version of the CLI you want to upgrade to and your CPU architecture, download one of the `.zip` files available on this page.
```

>     For example, if you wanted to install v1.0.0 of the Astro CLI on a Windows Machine with an AMD 64 architecture, you would download `astro_1.0.0-converged_windows_amd64.zip`.

Reviewer 2: ```suggestion
    For example, if you wanted to upgrade to v1.0.0 of the Astro CLI on a Windows machine with an AMD 64 architecture, you would download `astro_1.0.0-converged_windows_amd64.zip`.
```

3. Run the following command to unzip the executable:

    ```sh
    tar -xvzf .\astrocli.tar.gz
    ```

> 4. Save the new `astro.exe` file that you just unzipped in a secure location on your machine and add its filepath in the Windows PATH environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).

Reviewer 2: ```suggestion
4. Save the new `astro.exe` file that you just unzipped in a secure location on your machine and add its filepath as a Windows PATH environment variable. For more information about configuring the PATH environment variable, read [Java documentation](https://www.java.com/en/download/help/path.html).
```

</TabItem>

<TabItem value="linux">

To upgrade to the latest version of the Astro CLI, run:

```sh
curl -sSL install.astronomer.io | sudo bash -s
```

</TabItem>

</Tabs>

## Uninstall the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

> To uninstall the Astro CLI, run the following command:

Reviewer 2: ```suggestion
To uninstall the Astro CLI on Mac, run:
```

```sh
brew uninstall astronomer/tap/astro
```

</TabItem>

<TabItem value="windows">

> To uninstall the Astro CLI:

Reviewer 2: ```suggestion
To uninstall the Astro CLI on Windows:
```

1. Remove the filepath for `astro.exe` from your Windows PATH environment variable.

> 1. Delete `astro.exe`.

Reviewer 2: ```suggestion
2. Delete the `astro.exe` file from your machine.
```

</TabItem>

<TabItem value="linux">

> To uninstall the Astro CLI, run the following command:

Reviewer 2: ```suggestion
To uninstall the Astro CLI on Linux, run the following command:
```

```sh
sudo rm /usr/local/bin/astro
```

</TabItem>

</Tabs>

## Migrate from `astrocloud` to `astro`

Complete this migration if all of the following are true:

- You are an Astro user.

> - You used a pre-1.0 version of the Astro CLI. For these versions, the CLI executable was `astrocloud`.

Reviewer 2: ```suggestion
- You're currently using the `astrocloud` CLI executable.
```

> 

Reviewer 2: ```suggestion

For more information on Astro CLI v1.0.0, see [Astro CLI Release Notes](cli-release-notes.md).

```

### Step 1: Uninstall the `astrocloud` Executable

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To uninstall the `astrocloud` executable, run the following command:

```sh
brew uninstall astronomer/cloud/astrocloud
```

</TabItem>

<TabItem value="windows">

1. Remove the filepath for `astrocloud.exe` from your Windows PATH environment variable.
2. Delete `astrocloud.exe`.

</TabItem>

<TabItem value="linux">

To uninstall `astrocloud`:

1. Remove the filepath for `astrocloud.exe` from your Linux PATH environment variable.

> 2. Delete `astrocloud.exe`.

Reviewer 2: ```suggestion
2. Delete the `astrocloud.exe` file from your machine.
```

---
sidebar_label: 'Configure the CLI'
title: 'Configure the Astro CLI'
id: configure-cli
description: Install, upgrade, and manage settings for the Astro CLI.
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import {siteVariables} from '@site/src/versions';

## Install the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

#### Prerequisites

To use the Astro CLI on Mac, you must have:

- [Homebrew](https://brew.sh/)
- [Docker Desktop](https://docs.docker.com/get-docker/) (v18.09 or higher).

#### Installation

To install the latest version of the Astro CLI, run the following command:

```sh
brew install astronomer/tap/astro
```

To install a specific version of the Astro CLI, specify the version you want to install at the end of the command:

```sh
brew install astronomer/tap/astro@<major.minor.patch-version>
```

If you specify only a major version, this command will install the latest minor/patch version available for that major version.

</TabItem>

<TabItem value="windows">

> #### Prerequisites

Author: @kwcanuck I adapted your CLI instructions update for this PR, could you please review this Windows setup to make sure I captured all of the changes that you and Constance discussed? Feel free to ignore the rest of the PR, thanks!

Reviewer 3: @jwitz the procedures do not currently match at all. For consistency, the procedures should be identical. I recommend copying my revision, or updating install-cli.md in the `Clarify-Windows-CLI-Installation-Instructions-(344)` branch to match your content.

Author: @kwcanuck There are some details with the CLI v1.0.0 installation that have changed from previous versions, such as where you download the CLI from. I updated these instructions to match your PR in terms of details related to running the executable, but some steps had to change for this latest release

To use the Astro CLI on Windows, you must have:

- [Docker Desktop](https://docs.docker.com/desktop/windows/install/) for Windows.

> - [Docker Engine](https://docs.docker.com/engine/install/) (v0.13.1 or higher).

Author: ```suggestion
- [Docker Engine](https://docs.docker.com/engine/install/) (v0.18.9 or higher).
```

- [WSL](https://docs.microsoft.com/en-us/windows/wsl/install) enabled on your local machine.
-  Windows 10 or Windows 11.

#### Installation

> 1. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astro-projects/astro-cli/releases). Based on your desired CLI version and the CPU architecture of your machine, download one of the `.zip` files available on this page.

Reviewer: Yes, once Astro CLI is released their won't be a command to install on windows. Windows users will need to download the binary from this page https://github.com/astro-projects/astro-cli/releases

    For example, to install v1.0.0 of the Astro CLI on a Windows Machine with an AMD 64 architecture, download `astro_1.0.0-converged_windows_amd64.zip`.

2. Run the following command to unzip the executable:

    ```sh
    tar -xvzf .\astrocli.tar.gz
    ```

> 3. Add the filepath for the directory containing `astro.exe` as a PATH environment variable. For example, if `astro.exe` was stored in `C:\Users\username\astrocloud.exe`, you would add `C:\Users\username` as your PATH environment variable. To learn more about configuring the PATH environment variable, see [Java documentation](https://www.java.com/en/download/help/path.html).

Reviewer 3: This text doesn't match what is provided in the updated `install-cli.md` file. The text should be consistent. What file should be updated?

Reviewer: @jwitz what does if `astro.exe` was stored in `C:\Users\username\astrocloud.exe` mean? why are we mentioning astrocloud here?

Author: Typo on the `astrocloud` front, but now I'm wondering about whether we're adding `astrocloud.exe` or its directory as the PATH

</TabItem>

</Tabs>

## Upgrade the CLI

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To upgrade the Astro CLI to the latest version, you can run the same command as you did to install the CLI for the first time:

```sh
brew install astronomer/tap/astro
```

</TabItem>

<TabItem value="windows">

To upgrade the Astro CLI on Windows:

1. Remove the existing `astro.exe` file on your machine.

2. Go to the [**Releases** page of the Astro CLI GitHub repository](https://github.com/astro-projects/astro-cli/releases). Based on the version of the CLI you want and your CPU architecture, download one of the `.zip` files available on this page.

     For example, if you wanted to upgrade to v1.0.0 of the Astro CLI on a Windows machine with an AMD 64 architecture, you would download `astro_1.0.0-converged_windows_amd64.zip`.

3. Run the following command to unzip the executable:

    ```sh
    tar -xvzf .\astrocli.tar.gz
    ```

> 4. Add the filepath for the directory containing the new `astro.exe` as a PATH environment variable. For example, if `astro.exe` was stored in `C:\Users\username\astrocloud.exe`, you would add `C:\Users\username` as your PATH environment variable. To learn more about configuring the PATH environment variable, see [Java documentation](https://www.java.com/en/download/help/path.html).

Reviewer: should this be `C:\Users\username\astro.exe`

- You are an Astro user.
- You're currently using the `astrocloud` CLI executable.

For more information on Astro CLI v1.0.0, see [Astro CLI Release Notes](cli-release-notes.md).

### Step 1: Uninstall `astrocloud`

<Tabs
    defaultValue="mac"
    values={[
        {label: 'Mac', value: 'mac'},
        {label: 'Windows', value: 'windows'},
        {label: 'Linux', value: 'linux'},
    ]}>
<TabItem value="mac">

To uninstall `astrocloud` on Mac, run:

```sh
brew uninstall astronomer/cloud/astrocloud
```

</TabItem>

<TabItem value="windows">

To uninstall `astrocloud` on Windows:

1. Remove the filepath for `astrocloud.exe` from your Windows PATH environment variable.
2. Delete `astrocloud.exe` from your machine.

</TabItem>

<TabItem value="linux">

To uninstall `astrocloud` on Linux:

1. Remove the filepath for `astrocloud.exe` from your Linux PATH environment variable.
2. Delete `astrocloud.exe` from your machine.

</TabItem>

</Tabs>

### Step 2: Install Astro CLI v1.0+

For more details, see [Install the CLI](configure-cli.md#install-the-cli.md).

### Step 3: Migrate Project Configurations (_Optional_)

> If you manually updated the `.astrocloud/config.yaml` file of an existing Astro project:

Reviewer: They need to run astro dev init regardless of if they manually updated the `.astrocloud/config.yaml`

Reviewer: Users also need to run `astro dev init` if they are moving from astro v0 to astro v1

1. In your terminal, open your Astro project.
2. Run `astro dev init` to generate a new `.astro` directory in your project. This subdirectory might be hidden in graphical file browsers. You can show hidden files using `⌘ + Shift + .` on Mac or by selecting **View > Hidden items** in Windows file explorer.

>     If the CLI prompts you about whether you want to create a project in a non-empty directory, enter `Yes`. The CLI will only create files that aren't yet in your directory. In this case, the only file that it will create is `.astro/config.yaml`.

Reviewer: it also creates `./astro/test_dag_integrity_default.py`
